pkg/service/pipelinerun.go

```diff
@@ -380,3 +382,47 @@ func (s *service) ListPipelineRunsByRequester(ctx context.Context, req *pb.ListP
 		PageSize:     int32(pageSize),
 	}, nil
 }
+
+type uploadPipelineRunInputsToMinioParam struct {
+	PipelineTriggerID string
+	ExpiryRuleTag     string
+	PipelineData      []map[string]any
+}
+
+func (s *service) uploadPipelineRunInputsToMinio(ctx context.Context, param uploadPipelineRunInputsToMinioParam) error {
+	logger, _ := logger.GetZapLogger(ctx)
+	minioClient := s.minioClient
```

Comment:
nit: not something you introduced but I don't love having the `MinIO` client as a direct dependency in the domain layer.

Reply:
I wonder why MinIO isn't in repository layer.
I think we should move MinIO client to repository layer. 
Is it same as what you think?

If yes, I will register a ticket for it.

Reply:
Exactly, that was my thought ➕